pkg/remediator/conflict/handler.go

```diff
@@ -41,6 +43,16 @@ type Handler interface {
 	HasConflictError(core.ID) bool
 }
 
+// Record a management conflict error, including log and metric.
+func Record(ctx context.Context, handler Handler, err status.ManagementConflictError, commit string) {
```

Comment:
Perhaps this should be a method of `Handler`?

Reply:
I considered that, but it seemed weird. None of the other methods take a context. AddConflictError doesn't need the commit. And the handler doesn't otherwise "handle" metrics.

It could replace AddConflictError, but handler.Handle and handler.Record seem like generic names that would be weird with 4 arguments.

It's just sort of a helper function, not a method on the handler...

Reply:
I guess my feeling is that if the handler's reason for existing included logging and metrics it would compose things that help with that, right... like a logger or a metrics client? But we don't really do that in this code base. They all use package functions with global state. So it feels weird to have a handler method that calls both package functions and a handler method.

I can smell the code smell, I just can't put my finger on exactly why it smells.

Reply:
Maybe this is a better argument: Handler is currently just a type-safe & thread-safe OrderedMap. Adding a Record method that takes context and commit makes it messier and expands the scope and interface of the abstraction into something larger and nebulous. This is why coming up with a good name for that method is hard.